From d35ec59cf727c302cd499cc0d6efffd9efdcdce3 Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Thu, 27 Nov 2025 16:33:04 +0100
Subject: dmesg: fix const qualifier warnings in parse_callerid

Fix const qualifier discarded warnings in parse_callerid().
These warnings are reported by gcc 15 which defaults to the C23 standard.

The strchr() and strstr() functions return pointers into const strings,
so the receiving variables must be declared as const char *.

Signed-off-by: Karel Zak <kzak@redhat.com>
(cherry picked from commit 40e6850d2b2ba2492d5f3e75656cb9bcce2806bb)
---
 sys-utils/dmesg.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sys-utils/dmesg.c b/sys-utils/dmesg.c
index cccbd4067..cdf23b021 100644
--- a/sys-utils/dmesg.c
+++ b/sys-utils/dmesg.c
@@ -844,8 +844,7 @@ static const char *parse_callerid(const char *p_str, const char *end,
 	const char *p_after;
 	const char *p_next;
 	size_t cid_size;
-	char *p_scn;
-	char *p_cid;
+	const char *p_scn, *p_cid;
 
 	/* Check for PRINTK_CALLER prefix, must be before msg text */
 	p_cid = strstr(p_str, DMESG_CALLER_PREFIX);
-- 
2.51.1

